### PR TITLE
Mp s1 t5 build stage dropdown

### PR DESCRIPTION
**Pull request by elstevens23**
Created and populated stages dropdown on Edit Applicant page with stages from the database.

### CODE REVIEW

**Review by jennymegan — CHANGES_REQUESTED** [review]
just a quick one - I haven't looked all the way through but I think the list is supposed to include all the substages too (like in Booked In there are 3 sub stages etc)

File: src/Controllers/ApplicationFormController.php [hunk 1]
```diff
@@ -42,6 +42,7 @@ public function __invoke(Request $request, Response $response, array $args)
                     'data' => [
                         'cohorts' => $this->applicationFormModel->getCohorts(),
                         'hearAbout' => $this->applicationFormModel->getHearAbout(),
+                        'stages' => $this->applicationFormModel->getStages(),
```
Comment (ellie-r) [inline src/Controllers/ApplicationFormController.php:45]: should this be on the stage model?

Comment (ellie-r) [reply src/Controllers/ApplicationFormController.php:45]: should this use the $this->stageModel->getStageTitles() ?

File: templates/applicantForm.phtml [hunk 2]
```diff
@@ -18,6 +18,13 @@
 <?php require 'partials/layout/nav.phtml'; ?>
         <h3>Edit Applicant</h3>
         <form>
+            <section class="stagesDropdown">
+                <label for="stageTitle">Choose Applicants' Stage: </label>
```
Comment (ellie-r) [inline templates/applicantForm.phtml:22]: should this be Applicant's as it's about a single applicant

File: src/Models/StageModel.php [hunk 3]
```diff
@@ -60,6 +60,21 @@ public function createStage(StageEntity $stageEntity): bool
...
 
+    /**
+     * Gets all the stage titles from the database
+     *
+     * @return array
+     */
+    public function getStageTitles(): array
+    {
+        $query = $this->db->prepare(
+            'SELECT `id`, `title` FROM `stages`;'
```
Comment (ellie-r) [inline src/Models/StageModel.php:71]: does this need to be on a new line?

File: src/Models/StageModel.php [hunk 4]
```diff
@@ -216,6 +231,20 @@ public function getOptionsByStageId(int $stageId): array
...
+
+    /**
+     * Gets all the ids and options from the options table in the db
+     *
+     * @return array
+     */
+    public function getStageOptions(): array
+    {
+        $query = $this->db->prepare(
+            'SELECT `id`, `option`, `stageId` FROM `options`;'
```
Comment (ellie-r) [inline src/Models/StageModel.php:243]: does this need to be on a new line

File: src/Models/StageModel.php [hunk 5]
```diff
@@ -60,6 +60,21 @@ public function createStage(StageEntity $stageEntity): bool
...
+    public function getStageTitles(): array
+    {
+        $query = $this->db->prepare(
+            'SELECT `id`, `title` FROM `stages`;'
+        );
+        $query->execute();
+        return $query->fetchAll();
+    }
+
+
```
Comment (ellie-r) [inline src/Models/StageModel.php:77]: delete this line? so only one empty line between the methods

File: src/Models/ApplicationFormModel.php [hunk 6]
```diff
@@ -36,4 +36,17 @@ public function getHearAbout()
         $result = $query->fetchAll();
         return $result;
     }
+
+    /**
+     * Gets all the sages from the database.
+     *
+     * @return array $result is data from the stages fields.
+     */
+    public function getStages()
```
Comment (ellie-r) [inline src/Models/ApplicationFormModel.php:45]: is this method needed as theres the getStageTitles() method now

**Review by ellie-r — CHANGES_REQUESTED** [review]

File: src/Models/ApplicationFormModel.php (hunk 6 above)
Comment (elstevens23) [reply src/Models/ApplicationFormModel.php:45]: Took it out completely as it wasn't needed :)

File: src/Controllers/ApplicationFormController.php (hunk 1 above)
Comment (elstevens23) [reply src/Controllers/ApplicationFormController.php:45]: No longer necessary so we took this out.

File: src/Models/StageModel.php (hunk 3 above)
Comment (elstevens23) [reply src/Models/StageModel.php:71]: nope - we fixed it 👍

File: src/Models/StageModel.php (hunk 4 above)
Comment (elstevens23) [reply src/Models/StageModel.php:243]: now all on one line

File: templates/applicantForm.phtml (hunk 2 above)
Comment (elstevens23) [reply templates/applicantForm.phtml:22]: good point u meanie - done it

File: src/Models/StageModel.php (hunk 5 above)
Comment (ellie-r) [reply src/Models/StageModel.php]: missed this one

**Review by ellie-r — CHANGES_REQUESTED** [review]
you silly billys

**Review by ellie-r — APPROVED** [review]

**Review by jennymegan — APPROVED** [review]

**Pull request merged**